### ISSUE TITLE
Type guard doesn't work on import

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#12081]
**TypeScript Version:**  nightly (2.1.0-dev.20161107)

**Code**

```ts
// @strictNullChecks: true

// @Filename: /node_modules/@types/foo/index.d.ts
declare const foo: { x: number } | undefined;
export default foo;

// @Filename: /a.ts
import foo from "foo";
foo && foo.x;
```

**Expected behavior:**

No error.

**Actual behavior:** 

`error TS2532: Object is possibly 'undefined'.`

**Comment by andy-ms** [comment]
Fixed by #16658.

**Comment by mgroenhoff** [comment]
Awesome. Thanks! 👍
